components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -2803,6 +2803,44 @@ public IdentityProvider getIdPByRealmId(String realmId, int tenantId, String ten
 
     }
 
+    /**
+     * Get the enabled IDP of the given realm id.
+     *
+     * @param realmId       Realm ID of the required identity provider.
+     * @param tenantId      Tenant ID of the required identity provider.
+     * @param tenantDomain  Tenant domain of the required identity provider.
+     * @return              Enabled identity provider of the given realm id.
+     * @throws IdentityProviderManagementException Error when getting the identity provider.
+     * @throws SQLException                        Error when executing SQL query.
+     */
+    public IdentityProvider getEnabledIdPByRealmId(String realmId, int tenantId, String tenantDomain)
+            throws IdentityProviderManagementException {
+
+        Connection dbConnection = IdentityDatabaseUtil.getDBConnection(false);
```

Comment:
Better to initialize the connection inside the parentheses of the try block

Reply:
[bcbb83d](https://github.com/wso2/carbon-identity-framework/pull/4991/commits/bcbb83da651df7827aece0ed28b33f5166e59e0f)